Core/tests/Unit/RequestWrapperTest.php

```diff
@@ -668,6 +669,81 @@ public function testPassingInRetryOptions()
         $this->assertTrue($retryFunctionCalled);
         $this->assertTrue($retryListenerCalled);
     }
+
+    /**
+     * Test to verify auth related headers fetching and updation if
+     * credential fetcher implements UpdateMetadataInterface too, else
+     * going ahead with access_code fetching followed by auth header
+     * creation and updation.
+     */
+    public function testFetchingCredentialAsAuthHeader()
+    {
+        $credentialsFetcher = $this->prophesize(FetchAuthTokenInterface::class);
+        $credentialsFetcher->willImplement(UpdateMetadataInterface::class);
+
+        $accessToken = 'abc';
+        $credentialsFetcher->updateMetadata(Argument::cetera())->willReturn([
```

Comment:
What is the difference between `Argument::cetera` and `Argument::any`?

Reply:
Argument::cetera returns true for all the upcoming arguments in the argument list. For eg:

`$prophecy->method(Argument::type('string), Argument::cetera())` will match true for:
1) `$propehcy->method('hello')`
2) `$propehcy->method('hello', ['foo' => 'bar'])`
3) `$propehcy->method('hello', 'world')`
4) `$propehcy->method('hello', 'world', 'the' , 'weather', 'is', 'nice')`
5) `$propehcy->method('hello', true, false, true, [1, 2, 3])`

Where as Argument::any matches for a single argument of any type:
`$prophecy->method(Argument::type('string), Argument::any())` will match true for:
1) `$propehcy->method('hello')`
2) `$propehcy->method('hello', ['foo' => 'bar'])`
3) `$propehcy->method('hello', 'world')`

It will match false for:
1) `$propehcy->method('hello', 'world', 'the' , 'weather', 'is', 'nice')`
2) `$propehcy->method('hello', true, false, true, [1, 2, 3])`

Reply:
Would it be possible to narrow down the call a bit?

Reply:
The method is called with library preset headers and httphandler without any interference of user supplied `$options`. IMO it doesn't improves testing in checking them because in scenarios:

**Option1**: If we try to hardcode and check for the library set headers, the tests would start failing as soon as preset headers get changed or the keys change.
```php
$credentialsFetcher->updateMetadata(Argument::that(function ($arg) {
    return is_array($arg) && array_key_exists('x-goog-api-client', $arg) && array_key_exists('User-Agent', $arg);
}), Argument::any(), Argument::type('callable'))
            ->shouldBeCalledTimes(1)
            ->willReturn(['authorization' => ['Bearer ' . $accessToken]]);
```

**Option2**: If we just check the types of the argument then it would be replicating the function signature check which already happens while it's invocation implicitly.
```php
$credentialsFetcher->updateMetadata(Argument::type('array'), Argument::any(), Argument::type('callable'))
            ->shouldBeCalledTimes(1)
            ->willReturn(['authorization' => ['Bearer ' . $accessToken]]);
```

WDYT @bshaffer ?

Reply:
I am agreeing with you that in this particular case, the `cetera` call is probably the most readable and appropriate in this case.

Thanks for taking the time to explore other options! And please feel free to use them still if you prefer.